CMakeLists.txt

```diff
@@ -26,6 +26,17 @@ message("module path = ${CMAKE_MODULE_PATH}")
 file(GLOB source "source/*.cpp" "source/*.hpp" "source/*.h")
 if (APPLE)
 	set(OBJCPP "source/AppleUtilities.h" "source/AppleUtilities.mm")
+    # Detect whether we are on an Intel or ARM Mac
+    execute_process(COMMAND uname -m RESULT_VARIABLE result OUTPUT_VARIABLE OSX_ARCHITECTURE OUTPUT_STRIP_TRAILING_WHITESPACE)

```

Comment:
We should not be setting this.

Reply:
Why it's bad? I'm getting this strange issue and the fix solved it.
<img width="843" alt="Screenshot 2024-01-09 at 19 19 35" src="https://github.com/Ravbug/UnityHubNative/assets/6708914/05b48630-1555-4565-ae81-3d13f01d9874">

Reply:
The issue is we want to make fat binaries (ARM + x86) in a single executable so that both intel and apple silicon macOS users can run it. Setting `CMAKE_OSX_ARCHITECTURES` to a single architecture prevents creating fat binaries. 

Instead of `$(ARCHS_STANDARD)`, you can try setting it to `"x86_64;arm64"` and see if that also resolves the issue.

Reply:
Ah, I hadn't considered the possibility of fat binaries here. I gave it a try, and it worked. At least, the build was successful and it functioned properly on x64. I haven't been able to test it on ARM yet, as I haven't upgraded yet.